app/controllers/conference_registrations_controller.rb

```diff
@@ -20,6 +20,11 @@ def new
       redirect_to edit_conference_conference_registrations_path(@conference.short_title)
     end
 
+    if @conference.registration_limit_exceeded?
```

Comment:
Ah, good point!

Nevertheless, I'd add a validation to the registration model so it's checked on creation. Either a custom validation, sth like this:

``
validate :registration_limit_not_exceeded
 
  def :registration_limit_not_exceeded
    if registration_limit_exceeded
      errors.add(:base, "Registration limit exceeded.")
    end
  end
``

Or you could also have a look at http://apidock.com/rails/ActiveRecord/Validations/ClassMethods/validates_associated (though, never used it)

Furthermore you should hide the registration button on the front page and the splash page. Showing the button and simply redirect back to the root_path is not a good user experience from my point of view.

If we want to keep this code in the controller needs to be discussed, probably it makes sense for the case the link is distributed to social media etc...